policy_parser.py

```diff
@@ -14,35 +14,107 @@ def __init__(self, msg=None, policy=None):
 
 
 class PolicyParser(object):
+    """
+    Parser for Kubernetes NetworkPolicy API objects.
+
+    They are defined by the NetworkPolicySpec in the /apis/networking API
+    group.
+    """
     def __init__(self, policy):
-        self.namespace = policy["metadata"]["namespace"]
+        """
+        Create a Parser for a Kubernetes NetworkPolicy API object.
+
+        Returns the parser.
+        Throws PolicyError if the Policy is invalid.
+        """
         self.policy = policy
+        self.namespace = self.policy["metadata"]["namespace"]
+
+    def calculate_pod_selector(self):
+        """
+        Generate the Calico pod selector for this Policy.
+
+        Returns the pod selector in the Calico datamodel format.
+        Throws PolicyError if the Policy is invalid or unsupported.
```

Comment:
Looks like it does, but to be specific: Throws a PolicyError if the policy's `podSelector` is invalid or unsupported.

In practice, this should _never_ happen.